lib/modules/userInfo.js

```diff
@@ -142,6 +126,29 @@ addModule('userInfo', function (module, moduleID) { $.extend(true, module, {
 		var utctime = data.created_utc;
 		var d = new Date(utctime * 1000);
 		data.name = data.name.toLowerCase();
+
+		var header = $('<div />');
+		header.append('<a href="/user/' + escapeHTML(data.name) + '">/u/' + escapeHTML(data.name) + '</a> (<a href="/user/' + escapeHTML(data.name) + '/submitted/">Links</a>) (<a href="/user/' + escapeHTML(data.name) + '/comments/">Comments</a>)');
+		
+		// Add friend button to header.
+		RESUtils.getUserInfo(function(me) {
+			var friendButton, myID = 't2_' + me.data.id;
+			if (data.is_friend) {
+				friendButton = '<span class="fancy-toggle-button toggle" style="display: inline-block; margin-left: 12px;"><a class="option active remove" href="#" tabindex="100" onclick="return toggle(this, unfriend(\'' + data.name + '\', \'' + myID + '\', \'friend\'), friend(\'' + data.name + '\', \'' + myID + '\', \'friend\'))">- friends</a><a class="option add" href="#">+ friends</a></span>';
+			} else {
+				friendButton = '<span class="fancy-toggle-button toggle" style="display: inline-block; margin-left: 12px;"><a class="option active add" href="#" tabindex="100" onclick="return toggle(this, friend(\'' + data.name + '\', \'' + myID + '\', \'friend\'), unfriend(\'' + data.name + '\', \'' + myID + '\', \'friend\'))">+ friends</a><a class="option remove" href="#">- friends</a></span>';
+			}
+			header.append(friendButton);
+			def.notify(header, 'Got friend status, loading details...');
+			header.find('.toggle').click(function (e) {
+				if (e.target.tagName !== 'A') return;
+				// Expire the cache to re-check friend status.
+				RESUtils.cache.expire({key: 'RESUtils.cache.-user-' + data.name + '-about-json'});
+			});
+		});
+
+		def.notify(header, 'Getting friend status...');
```

Comment:
Yep, correct.
The one on this line would do something, but it's not noticeable since the promise is resolved immediately after (on [line 185](https://github.com/honestbleeps/Reddit-Enhancement-Suite/pull/2505/files#diff-a4534d46d42c0c1d56651e222f88c168R185)).
The one on [line 142](https://github.com/honestbleeps/Reddit-Enhancement-Suite/pull/2505/files#diff-a4534d46d42c0c1d56651e222f88c168R142) always occurs after the promise is resolved, so it does nothing.